sdk/appconfiguration/Azure.Data.AppConfiguration/tests/ConfigurationLiveTests.cs

```diff
@@ -947,14 +955,48 @@ await foreach (Page<ConfigurationSetting> page in service.GetConfigurationSettin
                 matchConditionsList.Add(matchConditions);
             }
 
-            foreach (MatchConditions matchConditions in matchConditionsList)
+            int pagesCount = 0;
+
+            await foreach (Page<ConfigurationSetting> page in service.GetConfigurationSettingsAsync(selector).AsPages(matchConditionsList))
```

Comment:
Ah, interesting.   so not in the client or options just passed when getting the set of pages.   I like this!